pkg/gofr/datasource/opentsdb/interface.go

```diff
@@ -7,6 +7,12 @@ import (
 	"time"
 )
 
+// HTTPClient is an interface that wraps the http.Client's Do method.
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+//nolint:unused // connection interface defines all the methods that needs to be mocked for testing purposes.
 type connection interface {
```

Comment:
```suggestion
// connection interface defines all the methods that needs to be mocked for testing purposes.
//
//nolint:unused
 type connection interface {
```

Reply:
But, you shouldn't have to use `//nolint:unused`

So either you force validating its use with something like

var _ connection = Mockconnection

Reply:
But then, if you need the `unused`, why would we need the interface?

Reply:
This interface is used to mock methods for testing. After you have pointed out, I’m thinking of redesigning it so both the tests and the OpenTSDB client use the same interface instead of having a separate dummy one. It’ll make things cleaner and more consistent.

Reply:
So, It is basically, because I need to test healthcheck method ...so I am using this interface to mock the connection that is to be returned when dialtimeout function is called.

Reply:
OK. Thanks, could you add this explanation in the `//nolintnolint:unused // blah blah` format

Maybe it could be addressed later.

Reply:
Ok.